File: testdata/jennies/rawtypes/disjunctions_of_scalars_and_refs/PHPRawTypes/src/DisjunctionsOfScalarsAndRefs/MyRefB.php
<?php

namespace Grafana\Foundation\DisjunctionsOfScalarsAndRefs;

class MyRefB implements \JsonSerializable
{
    public int $bar;

    /**
     * @param int|null $bar
     */
    public function __construct(?int $bar = null)
    {
        $this->bar = $bar ?: 0;
    }

    public function equals(mixed $other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }
    
        if ($this->bar !== $other->bar) {
            return false;
        }
    
        return true;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{bar?: int} $inputData */
        $data = $inputData;
        return new self(
            bar: $data["bar"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->bar = $this->bar;
        return $data;
    }
}

File: testdata/jennies/rawtypes/disjunctions_of_scalars_and_refs/JavaRawTypes/disjunctions_of_scalars_and_refs/MyRefB.java
package disjunctions_of_scalars_and_refs;

import java.util.Objects;

public class MyRefB {
    public Long bar;
    public MyRefB() {
        this.bar = 0L;
    }
    public MyRefB(Long bar) {
        this.bar = bar;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof MyRefB)) return false;
        MyRefB o = (MyRefB) other;
        if (!Objects.equals(this.bar, o.bar)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.bar);
    }
}
